apiserver/facades/agent/instancemutater/instancemutater.go

```diff
@@ -79,14 +79,16 @@ func NewInstanceMutaterAPI(st InstanceMutaterState,
 
 	getAuthFunc := common.AuthFuncForMachineAgent(authorizer)
 	return &InstanceMutaterAPI{
-		ModelMachinesWatcher: common.NewModelMachinesWatcher(st, resources, authorizer),
-		UnitsWatcher:         common.NewUnitsWatcher(st, resources, getAuthFunc),
-		LifeGetter:           common.NewLifeGetter(st, getAuthFunc),
-		st:                   st,
-		model:                model,
-		resources:            resources,
-		authorizer:           authorizer,
-		getAuthFunc:          getAuthFunc,
+		ModelMachinesWatcher: common.NewModelMachinesWatcherWithAuthFunc(st, resources, func() bool {
```

Comment:
My problem with this is that it looks like a general machine could ask to watch all the machines in a model. Surely the instance mutater is only interested in its own machines?

Reply:
@howbazaar can you add a bit more clarification on this... I think I'm missing something here.